### ISSUE TITLE
More thorough logging needed throughout

### ISSUE THREAD

**Issue by wfarner** [docker/infrakit#269]
From comment in #83:

> We really need to enhance the logging support. First of all, all logging should have timestamp. 2nd, the current code has little or no log, it is really hard to debug things when stuff goes wrong.

I concur with this statement that we should make a pass through major components to ensure that debug-level logging is more thorough.

**Comment by linsun** [comment]
Thank you @wfarner for opening this!

**Comment by wfarner** [comment]
@linsun i'm going to take a pass at this over the next few days.  If there are any particular bodies of  code where you would like to see more logging, feel free to drop a list here!

**Comment by linsun** [comment]
@wfarner Thanks for asking! I would like to see log finest level for entry/exit into any method, that is helpful when troubleshooting things.  I would like to see info level logs associated to key actions, like group watch request received, group is being destroyed, group is being created. I would like to see appropriate error and warning message in the logs.  When there is something wrong, would be great to see where the error was first occurred, whether this is a user error where user can potentially correct in group.json or plugin error or infrakit code error.  I recently tried rolling upgrade, but it didn't work for me, there was not much log at all other than it did detect that I requested an increase of 1 more instance number, but it also keeps saying "Waiting for scaler to quiesce"...
